components/centraldashboard/app/k8s_service.ts

```diff
@@ -24,12 +72,61 @@ export class KubernetesService {
   /** Retrieves the list of events for the given Namespace from the Cluster. */
   async getEventsForNamespace(namespace: string): Promise<k8s.V1Event[]> {
     try {
-      const {body} = await this.k8sApi.listNamespacedEvent(namespace);
+      const {body} = await this.coreAPI.listNamespacedEvent(namespace);
       return body.items;
     } catch (err) {
       console.error(
           `Unable to fetch Events for ${namespace}:`, err.body || err);
       return [];
     }
   }
+
+  /**
+   * Obtains cloud platform information from cluster Nodes,
+   * as well as the Kubeflow version from the Application custom resource.
+   */
+  async getPlatformInfo(): Promise<PlatformInfo> {
+    const info: PlatformInfo = {
+      provider: 'other://',
+      providerName: 'other',
+      kubeflowVersion: 'unknown'
+    };
+    // Invokes requests concurrently. A generic rejection handler is added to
+    // ensure that both Promises have a rejection handler when the first await
+    // is encountered to avoid any unhandled rejections.
+    const rejectionHandler = (): void => null;
+    const listNodePromise = this.coreAPI.listNode();
+    listNodePromise.catch(rejectionHandler);
+    const listApplicationsPromise =
+        this.customObjectsAPI.listNamespacedCustomObject(
+            APP_API_GROUP, APP_API_VERSION, this.namespace, APP_API_NAME);
+    listApplicationsPromise.catch(rejectionHandler);
+
+    // Gets the providerID from the first node
+    try {
+      const {body} = await listNodePromise;
+      const provider = body.items.map((n) => n.spec.providerID).find(Boolean);
+      if (provider) {
+        info.provider = provider;
+      }
+      info.providerName = info.provider.split(':')[0];
+    } catch (err) {
+      console.error('Unable to fetch Node information:', err.body || err);
+    }
+
+    // Gets the first Application entity for the namespace
+    try {
+      const response = await listApplicationsPromise;
+      const body = response.body as V1BetaApplicationList;
+      if (body.items && body.items.length && body.items[0].spec &&
+          body.items[0].spec.descriptor &&
+          body.items[0].spec.descriptor.version) {
+        info.kubeflowVersion = body.items[0].spec.descriptor.version;
```

Comment:
While I agree that this is a more succinct solution, it's also much more difficult to read. There are alternatives that would allow for Java 8 style Optionals (https://github.com/bromne/typescript-optional), and it may make sense to include them in the future but I tend to favor the [Python-zen guidelines](https://www.python.org/dev/peps/pep-0020/#id3) to avoid favoring brevity over readability.